File: search-service/src/main/java/com/stackroute/eplay/search/domain/Movie.java
package com.stackroute.eplay.search.domain;

import java.time.LocalDate;
import java.util.List;

import org.springframework.data.annotation.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

//Pojo for Movie
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Movie {
	@Id
	private int id;
	private String name;
	private String poster;
	private String genre;
	private String language;
	private String reviews;
	private String cast;
	private String description;
	private int rating;
	private int duration;
	private List<MovieEvent> movieEvents;
	private LocalDate releaseDate;
	private List<Theatre> theatres;

}

File: Front-end/src/app/movie.ts
import {Theatre} from "./theatre";

export class Movie {
  id: number;
  name: string;
  poster: string;
  genre: string;
  language: string;
  reviews: string;
  cast: string;
  description: string;
  rating: number;
  duration:number;
  releaseDate: string
  movieEvents: string

  theatre: Array<Theatre>;
}
